cfgov/unprocessed/js/modules/multi-select.js

```diff
@@ -0,0 +1,494 @@
+          var item = self.list.querySelector( 'li[data-option="' + option.value + '"]' );
+
+          $( item ).classList.add( 'filter-match' );
+        } );
+      } else {
+        this.noResults();
+      }
+    },
+
+    noResults: function() {
+      this.list.classList.add( 'no-results' );
+      this.list.classList.remove( 'filtered' );
+    },
+
+    bindEvents: function() {
+      var self = this;
+      var inputs = this.list.querySelectorAll( 'input' );
+
+      var things = $( '.form-l' ).querySelectorAll( '*' );
+
+      $.bind( this.search, {
+        input: this.evaluate.bind( this ),
+        focus: this.open.bind( this ),
+        blur: function( event ) {
+          if ( !self.skipBlur ) {
+            self.close();
+          }
+        },
+        mousedown: function() {
+          if ( self.fieldset.getAttribute( 'visibility' ) === 'hidden' ) {
+            self.open();
+          }
+        },
+        keydown: function( event ) {
+          var key = event.keyCode;
+
+          if ( self.fieldset.getAttribute( 'visibility' ) === 'hidden' ) {
+            self.open();
+          }
+
+          if ( key === 13 ) {
+            event.preventDefault();
+            self.next();
+          } else if ( key === 27 ) {
+            self.close();
+          } else if ( key === 38 ) {
+            event.preventDefault();
+            self.previous();
+          } else if ( key === 40 ) {
+            event.preventDefault();
+            self.next();
+          } else if ( key === 9 && !event.shiftKey ) {
+            if ( self.fieldset.getAttribute( 'visibility' ) === 'visible' ) {
+              self.close();
+            }
+          }
+        }
+      } );
+
+      $.bind( $( this.list ), {
+        mousedown: function( event ) {
+          self.skipBlur = true;
+        },
+        keydown: function( event ) {
+          var key = event.keyCode;
+          var checked = event.target.checked;
+          console.log( key );
+
+          if ( key === 13 ) {
+            event.preventDefault();
+            if ( checked ) {
+              event.target.checked = false;
+            } else {
+              event.target.checked = true;
+            }
+            $.fire( $( event.target ), 'change', {} );
+          } else if ( key === 27 ) {
+            self.search.focus();
+            self.close();
+          } else if ( key === 38 ) {
+            event.preventDefault();
+            self.previous();
+          } else if ( key === 40 ) {
+            event.preventDefault();
+            self.next();
+          }
+        }
+      } );
+
+      forEach( inputs, function( i ) {
+        $.bind( $( inputs[i] ), {
+          change: function( event ) {
+            self.updateSelections( event.target.id );
+            self.resetSearch();
+          }
+        } );
+      } );
+    }
+  };
+
+  // Static methods/properties
+  _.all = [];
+
+  _.FILTER_CONTAINS = function ( text, input ) {
```

Comment:
I guess also because `FILTER_CONTAINS` will have different values depending on what the values of `text` and `input` passed into it, so referencing `FILTER_CONTAINS` will not always return the same value.